netket/optimizer/qgt/qgt_jacobian.py

```diff
@@ -0,0 +1,290 @@
+# Copyright 2021 The NetKet Authors - All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional, Union
+
+import jax
+
+from netket.utils import timing
+from netket.utils.api_utils import partial_from_kwargs
+from netket import jax as nkjax
+from netket.nn import split_array_mpi
+
+from .qgt_jacobian_dense import QGTJacobianDenseT
+from .qgt_jacobian_pytree import QGTJacobianPyTreeT
+from .qgt_jacobian_common import (
+    sanitize_diag_shift,
+    to_shift_offset,
+    rescale,
+)
+
+
+@timing.timed
+def QGTJacobian_DefaultConstructor(
+    apply_fun,
+    parameters,
+    model_state,
+    samples,
+    pdf=None,
+    *,
+    dense: bool,
+    mode: Optional[str] = None,
+    holomorphic: Optional[bool] = None,
+    diag_shift: float = 0.0,
+    diag_scale: float = 0.0,
+    chunk_size: Optional[int] = None,
```

Comment:
this function can be used by pretty much anyone, even not really using variational states, to build a QGT.

I like this because it unties our QGTs from our variational states.